runtime/src/main/scala/akka/grpc/GrpcProtocol.scala

```diff
@@ -87,7 +87,7 @@ object GrpcProtocol {
       messageEncoding: Codec,
       /** Encodes a frame as a part in a chunk stream. */
       encodeFrame: Frame => ChunkStreamPart,
-      /** A shortcut to encode a data frame directly into a ByteString */
+      /** A shortcut to encode a data frame directly into a Response */
       encodeDataToResponse: (ByteString, immutable.Seq[HttpHeader], Trailer) => HttpResponse,
```

Comment:
@jrudolph I think Trailer could get some optimizations as well—having a cached instance for an empty Trailer, and avoiding creating a collection of 1 element and then immediately creating a new collection via map: https://github.com/akka/akka-http/blob/main/akka-http-core/src/main/scala/akka/http/scaladsl/model/Trailer.scala#L33

Reply:
For akka-grpc we are already using a cached instance here: https://github.com/jrudolph/akka-grpc/blob/efda1b1ff4a8fd9849be66a5cd8c66369f6027af/runtime/src/main/scala/akka/grpc/internal/GrpcResponseHelpers.scala#L32. For akka-http itself, you could also just omit the trailer when not needed.